substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -358,10 +363,9 @@ pub mod pallet {
 	/// The map from nominator stash key to their nomination preferences, namely the validators that
 	/// they wish to support.
 	///
-	/// Note that the keys of this storage map might become non-decodable in case the
-	/// account's [`NominationsQuota::MaxNominations`] configuration is decreased.
-	/// In this rare case, these nominators
-	/// are still existent in storage, their key is correct and retrievable (i.e. `contains_key`
+	/// Note that the keys of this storage map might become non-decodable in case the account's
```

Comment:
this sounds bad. maybe we can add a try-state check to ensure this is caught and provide instructions to clean up the undecodable storage if governance wants to reduce this MaxNominations? 

it should already be caught though when try-runtime tries to decode all state after running the migrations, but still extra layers with better err messages would be nice.

Reply:
Yeah this is a general issue related to when you any `Get` type used as `BoundedXXX` is decreased. IMO the solution, other than the checks you mentioned, is something like https://github.com/paritytech/polkadot-sdk/issues/1139.